File: 0070_Climbing Stairs/ClimbingStairsDP.cs
public class Solution {
    public int ClimbStairs(int n) {
        if(n < 2) return 1;
        int prev2 = 1; 
        int prev1 = 1;
        int curr = 0;
        for(int i=2;i<=n;i++){
            curr = prev2 + prev1;
            prev2 = prev1;
            prev1 = curr;
        }

        return curr;
    }
}

File: 0070_Climbing Stairs/ClimbingStairsDP.java
class Solution {
    public int climbStairs(int n) {
        if(n == 1) return 1;
        if(n == 2) return 2;
        
        int a = 1, b = 2;
        for(int i=3;i<=n;i++) {
            int c = a + b;
            a = b;
            b = c;
        }
        
        return b;
    }
}